olp-cpp-sdk-core/tests/cache/DefaultCacheImplTest.cpp

```diff
@@ -1447,4 +1449,53 @@ std::vector<OpenTestParameters> DefaultCacheImplOpenParams() {
 INSTANTIATE_TEST_SUITE_P(, DefaultCacheImplOpenTest,
                          testing::ValuesIn(DefaultCacheImplOpenParams()));
 
+TEST_F(DefaultCacheImplTest, ProtectedCacheIOErrorFallbackToReadOnly) {
+  SCOPED_TRACE("IOError fallback to read-only for protected cache");
+
+  // Create a writable directory with a subdirectory named "LOCK" to force
+  // LevelDB to fail when trying to create its lock file, triggering IOError.
+  // This exercises the fallback branch (lines 787-789) that retries opening
+  // in read-only mode after an IOError in write mode.
```

Comment:
the rest of the file contains pretty simple tests without numerous comments, maybe it make sense to stick with that stile. And it make sense to ensure this test is reaching the point you are interested in.
Not critical, just looks very different, feel free to keep as it is if you believe that's how it should be

Reply:
That is probably the reason of windows fails.
There was no need to make the test or change it, that was just a point of improvement